kensu/matplotlib/pyplot.py

```diff
@@ -0,0 +1,56 @@
+from matplotlib.pyplot import *
+import matplotlib.pyplot as plt
+from functools import partial
+
+from kensu.utils.helpers import eventually_report_in_mem, get_absolute_path
+
+
+def subplots(*args,**kwargs):
+    orig_return = plt.subplots(*args, **kwargs)
+
+    for i in range(len(orig_return[1])):
+        def add_inheritance(self, arg):
+            self.inheritance.append(arg)
+        orig_return[1][i].inheritance = []
+        orig_return[1][i].add_inheritance = partial(add_inheritance, orig_return[1][i])
+
+        from kensu.utils.wrappers import kensu_wrapper_and_report
+        for f in dir(orig_return[1][i]):
+            if f in ['bar','boxplot','hist','hist2d','plot','plot_date']:
+                exec("setattr(orig_return[1][i],f, kensu_wrapper_and_report(orig_return[1][i].%s,orig_return[1][i]))" % f)
```

Comment:
- orig_return[1][i] could be a variable, maybe `res_field=orig_return[1][i]`
- why not `orig_return[1][i].%s` -> `getattr(orig_return[1][i], f)` ? so it would not need this nasty live string as code interpolation ? like this:
```py
result_entry=orig_return[1][i] # you can then replace all usages above and below
orig_entry_field= getattr(res_entry, field_name)
setattr(res_field, field_name, kensu_wrapper_and_report(orig_field,result_entry)
```